addons/fix-costume-drag/addon.json

```
@@ -0,0 +1,53 @@
+{
+  "name": "Improved dragging of costumes and sounds",
+  "description": "When dragging a costume or sound, automatically scrolls the list and fixes a Scratch bug where the asset isn't placed as expected after scrolling.",
+  "credits": [
+    {
+      "name": "mysinginmonsters",
+      "link": "https://scratch.mit.edu/users/mysinginmonsters/"
+    }
+  ],
+  "settings": [
+    {
+      "name": "Auto-scroll speed",
+      "id": "scroll-speed",
+      "type": "select",
+      "description": "Drag a costume or sound at the top or bottom of the list to auto-scroll",
+      "potentialValues": [
+        {
+          "id": "none",
+          "name": "None"
+        },
+        {
+          "id": "slow",
+          "name": "Slow"
+        },
+        {
+          "id": "default",
+          "name": "Default"
+        },
+        {
+          "id": "fast",
+          "name": "Fast"
+        }
+      ],
+      "default": "default"
+    }
+  ],
+  "userscripts": [
+    {
+      "url": "userscript.js",
+      "matches": ["projects"]
+    }
+  ],
+  "userstyles": [
+    {
+      "url": "userstyle.css",
+      "matches": ["projects"]
+    }
+  ],
+  "tags": ["editor", "costumeEditor"],
```

```suggestion
  "tags": ["editor", "costumeEditor", "recommended"],
```
Maybe this could also be enabled by default?

---

I don't really want to understand what this addon is about today, so I suggest it gets `"enabledByDefault": true` if both of the following things happen:
- mxmou agrees
- There are open issues in the scratch-gui repo for the bug(s) this fixes

---

The positioning bug has an issue with no replies: https://github.com/scratchfoundation/scratch-gui/issues/5063

I didn't find one for auto-scrolling.

---

Why has that addon been in the works for 2 years and it's not added yet?

---

> Why has that addon been in the works for 2 years and it's not added yet?

It was finished 2 years ago but nobody has reviewed the PR yet.

---

I would like to merge it with the name "Improved dragging in editor," but then the sprite panel would be the only thing that isn't fixed and I can't figure it out for the life of me.

---

> Not sure why I specifically should agree

Maybe I should have said "doesn't disagree"